accepted/0000-mcp-server-auth.md

```diff
@@ -0,0 +1,123 @@
+# Authenticating Uyuni MCP server through the Uyuni API
+
+# Summary
+[summary]: #summary
+
+This RFC presents three architectural patterns for securely connecting the Uyuni MCP server to the Uyuni API. The core challenge is bridging an OAuth-based system with Uyuni's existing credential-based API. This document details three viable solutions: a "Service Account with Internal Authorization" pattern that requires no changes to Uyuni's codebase, an "OAuth Token Exchange" pattern that offers tighter integration at the cost of modifying Uyuni, and a "Per-User API Key" pattern, which would first require adding a new API key generation and management system to Uyuni. All designs aim to enable secure, user-delegated actions from an LLM-based agent.
+
+# Motivation
+[motivation]: #motivation
+
+To enable LLM-powered MCP agents to interface with Uyuni, a secure method for delegating user actions is required. Storing end-user credentials directly on the MCP server is not a viable or secure option, so this RFC defines architectural patterns that respect Uyuni's security model while enabling this new type of interaction.
+
+The expected outcome is a secure and auditable system where the MCP server can reliably execute actions against the Uyuni API. All actions will be authorized based on the end-user's actual permissions (RBAC) within Uyuni, providing a safe and effective bridge between conversational AI and the traditional API workflow.
+
+# Detailed design
+[design]: #detailed-design
+
+To solve the authentication and authorization challenge, this RFC presents three distinct patterns. The choice between them depends on the trade-off between implementation effort and the desired level of security and auditing granularity.
+
+---
+
+### Pattern 1: Service Account with Internal Authorization
+
+**Architecture:**
+
+1.  **User Authentication (OAuth)**: The end-user authenticates with the MCP proxy ecosystem via a standard OAuth 2.0 flow. The LLM agent receives a JWT access token representing the user.
+
+2.  **MCP Server Authentication**:
+    - A single, non-human **Service Account** (e.g., `mcp-service-account`) is created in Uyuni with a powerful role that includes the permissions necessary to execute any tool within the Uyuni MCP server.
```

Comment:
So would it exist a custom Uyuni role for the service that would be read-only access except for whatever is needed to run the available tools, correct?